### DISCUSSION TITLE
Add Fail Safe Detection

### DISCUSSION THREAD

**Discussion by FatheredPuma81** [valinet/ExplorerPatcher#1097] [Ideas]
Booted Windows 11 after an update to be greeted with Explorer.exe restarting every second. Had no idea what was going on for the first 10 minutes. Had to open msedge with Run, open CMD with Run, force kill Explorer with taskkill, and then hunt down this github to get the latest version which to my luck worked. I'm guessing many users have had to reinstall their entire OS because of this.

Would be lovely if there was some sort of failsafe that resets everything temporarily after say 30 Explorer restarts in a row.

**Comment by AlexAdasCca** [comment]
See the workaround here: https://github.com/valinet/ExplorerPatcher/discussions/1087

**Reply by FatheredPuma81** [reply]
Please leave bot.

**Reply by AlexAdasCca** [reply]
?No, I'm just linking answer to some repetitive questions. Perhaps this is a solution.

**Reply by FatheredPuma81** [reply]
I skimmed it and see nothing that adds a failsafe. Please elaborate.

**Reply by AlexAdasCca** [reply]
Using the latest version of ExplorerPatcher, then downloading ViveTool and disabling 26008830 may work.

**Reply by AlexAdasCca** [reply]
Microsoft says that:
![HWS2 WBTIKU))Q 7RTOA265](https://user-images.githubusercontent.com/116294917/225959576-59697910-567f-4594-be59-ddc5b623f400.png)

**Reply by AlexAdasCca** [reply]
> Booted Windows 11 after an update to be greeted with Explorer.exe restarting every second. Had no idea what was going on for the first 10 minutes. Had to open msedge with Run, open CMD with Run, force kill Explorer with taskkill, and then hunt down this github to get the latest version which to my luck worked. I'm guessing many users have had to reinstall their entire OS because of this.
> 
> Would be lovely if there was some sort of failsafe that resets everything temporarily after say 30 Explorer restarts in a row.
Try this please:
https://github.com/valinet/ExplorerPatcher/discussions/1087#discussioncomment-5346671

**Reply by FatheredPuma81** [reply]
> Using the latest version of ExplorerPatcher, then downloading ViveTool and disabling 26008830 may work.

May work for what?
> **This is only a temporary solution. Future system version updates may also result in the invalidation of relevant functions on the taskbar, and then it is necessary to use this tool or other methods again.** Moreover, I think it is necessary to disassemble the sequence and find the cause of the problem, otherwise it will be troublesome to disable some features after every important Windows update.

By your own admission this doesn't add a failsafe and is temporary until (my guess is) the next Windows update. I don't know what you're trying to achieve. The only _issue_ I have currently is the lack of a Volume bar when I change volume with my keyboard which I don't particularly care about.
> Microsoft says that:

Hey ultimately it's a suggestion. It's up to the developer if he believes it's worth his time to implement. If there's a high enough chance to just not boot Windows maybe it's not.

**Reply by topia** [reply]
Ultimately, binary patching to software is very fragile, except for public API.
For example, we can check the windows build version and stop patching it until devs verify it working, but I'm sure we don't want it.
Balancing between usefulness and safety is a very difficult problem.

Moreover, If I understand correctly, the crashing problem encountered recently is "another interface to implement".
https://github.com/valinet/ExplorerPatcher/commit/f9d702ebbf81699a94f1dd8a98ea5d3ef48c4583
You can't implement a fail-safe for the such problem before Microsoft releases it.

**Reply by topia** [reply]
> Would be lovely if there was some sort of failsafe that resets everything temporarily after say 30 Explorer restarts in a row.

Ah, now I have understood what you said. "detects if explorer restarted many times in a short period" and "stop patching at all to fight that problem".

**Reply by FatheredPuma81** [reply]
Yes exactly. Better to stop patching than to be stuck in a crash loop.

**Comment by AlexAdasCca** [comment]
This requires restarting the PC.

**Comment by FatheredPuma81** [comment]
https://github.com/valinet/ExplorerPatcher/commit/d7e5b7d3c965803d4d52aefd7fedefe11e3518c4
Thanks!
